src/majsoulrpa/_impl/browser.py

```diff
@@ -164,16 +165,34 @@ def __init__(
         initial_position = f"--window-position={initial_left},{initial_top}"
         proxy_server = f"--proxy-server=http://localhost:{proxy_port}"
         ignore_certifi_errors = "--ignore-certificate-errors"
-        options = [initial_position, proxy_server, ignore_certifi_errors]
+        options = [
+            initial_position,
+            proxy_server,
+            ignore_certifi_errors,
+        ]
         mute_audio_off = None if headless else ["--mute-audio"]
 
         self._context_manager = sync_playwright()
-        self._browser = self._context_manager.start().chromium.launch(
-            args=options,
-            ignore_default_args=mute_audio_off,
-            headless=headless,
-        )
-        self._context = self._browser.new_context(viewport=self._viewport_size)  # type: ignore[arg-type]
+
+        self._browser = None
+        if userdata_dir:
+            self._context = self._context_manager.start(
+            ).chromium.launch_persistent_context(
+                userdata_dir,
+                args=options,
+                ignore_default_args=mute_audio_off,
+                headless=headless,
+            )

```

Comment:
決まったコーディング規約があると思わず flake8 で lint するだけに留めていました。以降は ruff による format/lint を実施します。

ruff のユーザー設定は何もなく、リポジトリの pyproject.toml の設定した場合のフォーマット結果が一致しないため質問させてください。リポジトリの pyproject.toml の設定（line-length 79）の場合で @Apricot-S さんの結果と一致しません。black が使っている line-length 88 をコマンドライン引数に与えて明示的に指定すると @Apricot-S さんの結果と一致します。フォーマットの一致のさせかたになにか見落としはありますでしょうか？

```
❯ ruff format src/majsoulrpa/_impl/browser.py --config pyproject.toml --line-length 79 --diff
warning: The following rules may cause conflicts when used with the formatter: `COM812`, `ISC001`. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding them to the `ignore` configuration.
--- src/majsoulrpa/_impl/browser.py
+++ src/majsoulrpa/_impl/browser.py
@@ -176,8 +176,7 @@
 
         self._browser = None
         if userdata_dir:
-            self._context = self._context_manager.start(
-            ).chromium.launch_persistent_context(
+            self._context = self._context_manager.start().chromium.launch_persistent_context(
                 userdata_dir,
                 args=options,
                 ignore_default_args=mute_audio_off,

1 file would be reformatted
❯ ruff format src/majsoulrpa/_impl/browser.py --config pyproject.toml --line-length 88 --diff
warning: The following rules may cause conflicts when used with the formatter: `COM812`, `ISC001`. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding them to the `ignore` configuration.
--- src/majsoulrpa/_impl/browser.py
+++ src/majsoulrpa/_impl/browser.py
@@ -176,12 +176,13 @@
 
         self._browser = None
         if userdata_dir:
-            self._context = self._context_manager.start(
-            ).chromium.launch_persistent_context(
-                userdata_dir,
-                args=options,
-                ignore_default_args=mute_audio_off,
-                headless=headless,
+            self._context = (
+                self._context_manager.start().chromium.launch_persistent_context(
+                    userdata_dir,
+                    args=options,
+                    ignore_default_args=mute_audio_off,
+                    headless=headless,
+                )
             )
         else:
             self._browser = self._context_manager.start().chromium.launch(

1 file would be reformatted
```

Reply:
Ruff のドキュメントによる length limit は best effort attempt であるとのことがドキュメントに記述されていたため、完全な一致は考えず、ruff check により主要なエラーが出ないよう努めます。

> While the [line-too-long](https://docs.astral.sh/ruff/rules/line-too-long/) (E501) rule can be used alongside the formatter, the formatter only makes a best-effort attempt to wrap lines at the configured [line-length](https://docs.astral.sh/ruff/settings/#line-length). As such, formatted code may exceed the line length, leading to [line-too-long](https://docs.astral.sh/ruff/rules/line-too-long/) (E501) errors.

* ref: https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules

Reply:
すみません。フォーマット結果の違いについては私のミスでした
私の試したコード片ではインデントが1段足りなかったので79文字に収まっていましたが
修正したところsmlyさんと同じ結果になりました

- 79文字に収めようと`\`で改行してもRuffが一行にまとめてしまう
- ドット部分で改行するとかえって読みにくい
ので、79文字をはみ出すのは`# noqa: E501`で無視してもよいかもしれません
「一貫性にこだわりすぎるのは、狭い心の現れである」とも言いますし
smlyさんはどう思われますか?

Reply:
確認ありがとうございます。結果が一致するならば ruff format の結果を使いましょう👍
機械的にコード品質をあわせることができ、検査もできるので助かります。

`# noqa: E501` による無視が好みです。一旦この方法で差分を作成します。

Reply:
https://github.com/Apricot-S/majsoulrpa/pull/110/commits/a7047183b22137dc1d2c025f6b53f63d559378df にて対応